src/helpers.php

```diff
@@ -246,3 +246,37 @@ function is_countable($obj)
         return is_array($obj) || $obj instanceof Countable;
     }
 }
+
+if (! function_exists('oldValueDefaultOrFallback')) {
+    /**
+     * This function allows us to setup a default value in case there is an old value, but is forcelly null
+     * by Laravel middleware ConvertEmptyStringsToNull.
+     *
+     * @param  string  $field_name
+     * @param  mixed  $fallback
+     * @return mixed
+     */
+    function oldValueDefaultOrFallback($field, $fallback, $field_name = null, $default = null, $value = null)
+    {
+        $field_name = $field_name ?? $field['name'] ?? null;
+        $default = $default ?? $field['default'] ?? null;
+        $value = $value ?? $field['value'] ?? null;
+
+        // When you submit your field empty, internaly laravel will convert it to «null».
+        // Using old('field') with coalescing operator (??) we can't match it even if we have it in the old() session (as null).
+        // This checks that the value is null, but exists in the session old array, so it was submited empty|null, but exists!
+        if (! empty(session()->getOldInput())) {
+            if (is_null(old(square_brackets_to_dots($field_name)))) {
+                return $fallback;
+            }
+
+            return old(square_brackets_to_dots($field_name));
+        }
+
+        if (array_key_exists('value', $field)) {
+            return $value;
+        }
+
+        return $default ?? $fallback;
```

Comment:
Hi @pxpm, can't/shouldn't we use this instead?

```suggestion
        return $value ?? $default ?? $fallback;
```

So, if `array_key_exists('value', $field)`, it will be assigned above to the `$value`, on `$value = $value ?? $field['value'] ?? null;`.

And as it is right now, I think the code on `date_range.blade.php` will not work, because it may be sending the `$value` on the last arg, but it will never be used.

```
oldValueDefaultOrFallback($field, '', $field['name'][0], $start_default, $start_value ?? null)
```

`$start_value` will be assigned to `$value`, but since it doesn't exists on the `$field` array, it will never be reached.

Btw, I haven't tested a solution where this happens, I'm just assuming by reading the code.
Let me know if I'm "reading" something wrong.

Reply:
So hey! \o

I think you are rigth about the date_range. 

I've just refactored to use your proposed solution. 

Thanks @promatik